client/quota_test.go

```diff
@@ -56,7 +57,12 @@ func (cs *clientSuite) TestEnsureQuotaGroup(c *check.C) {
 		"group-name": "foo",
 		"parent":     "bar",
 		"snaps":      []interface{}{"snap-a", "snap-b"},
-		"max-memory": float64(1001),
+		"constraints": map[string]interface{}{
+			// meh this is exactly '1001' in the JSON request, but unless we
+			// parse it as a json.Number into an int64, it will show up typed as
+			// a float64, so this is easier to check in the test
+			"memory": 1001.0,
```

Comment:
(probably-ignore) This got me curious and it seems by using jsonutil.DecodeWithNumber we can get around this mostly, something like:
```go
diff --git a/client/quota_test.go b/client/quota_test.go
index c3076acea5..fe7d3f34df 100644
--- a/client/quota_test.go
+++ b/client/quota_test.go
@@ -20,6 +20,7 @@
 package client_test
 
 import (
+       "bytes"
        "encoding/json"
        "io/ioutil"
 
@@ -27,6 +28,7 @@ import (
 
        "github.com/snapcore/snapd/client"
        "github.com/snapcore/snapd/gadget/quantity"
+       "github.com/snapcore/snapd/jsonutil"
 )
 
 func (cs *clientSuite) TestCreateQuotaGroupInvalidName(c *check.C) {
@@ -50,7 +52,7 @@ func (cs *clientSuite) TestEnsureQuotaGroup(c *check.C) {
        body, err := ioutil.ReadAll(cs.req.Body)
        c.Assert(err, check.IsNil)
        var req map[string]interface{}
-       err = json.Unmarshal(body, &req)
+       err = jsonutil.DecodeWithNumber(bytes.NewReader(body), &req)
        c.Assert(err, check.IsNil)
        c.Assert(req, check.DeepEquals, map[string]interface{}{
                "action":     "ensure",
@@ -58,10 +60,7 @@ func (cs *clientSuite) TestEnsureQuotaGroup(c *check.C) {
                "parent":     "bar",
                "snaps":      []interface{}{"snap-a", "snap-b"},
                "constraints": map[string]interface{}{
-                       // meh this is exactly '1001' in the JSON request, but unless we
-                       // parse it as a json.Number into an int64, it will show up typed as
-                       // a float64, so this is easier to check in the test
-                       "memory": 1001.0,
+                       "memory": json.Number("1001"),
                },
        })
 }
```
but honestly not sure if it's really worth it

Reply:
ah nice trick, I couldn't manage to get that working but I suppose I wasn't using DecodeWithNumber in this test, I will make this change

Reply:
done